packages/dashboards/addon/templates/dashboards/dashboard/widgets/widget.hbs

```diff
@@ -131,7 +131,7 @@
         Run
       </button>
 
-      {{#if (and model.dashboard.canUserEdit model.hasDirtyAttributes)}}
+      {{#if (and model.dashboard.canUserEdit model.hasDirtyAttributes (or showSaveWidget (not model.isNew)))}}
```

Comment:
It looks like clicking `Save` already fires off the `runReport` action. Maybe there's a way to have the `saveWidget` route action wait for the `runReport` action to resolve first.

Reply:
Thanks for that, `transitionTo` does resolve after the `model` resolves so all we missed was a `return`. 🤦‍♂ 
It still however doesn't fix adding existing new reports with bad vis config since we don't go through the view route.